src/helpers/truncate.ts

```diff
@@ -1,11 +1,31 @@
 import { isPositiveInt } from "../validators/isNumber";
 
-export const truncate = (arg: string, limit: number, ellipsis = "...") => {
+export const truncate = (
+  arg: string,
+  limit: number,
+  ellipsis = "...",
+  position: "start" | "middle" | "end" = "end"
+) => {
   if (!isPositiveInt(limit)) return arg;
 
   const argArray = [...arg]; // convert string to array, emoji and unicode safe
 
   if (argArray.length <= limit) return arg;
 
-  return argArray.slice(0, limit).join("") + ellipsis;
+  switch (position) {
+    case "start":
+      return ellipsis + argArray.slice(-limit).join("");
+    case "middle": {
+      const startSliceLength = Math.floor(limit / 2);
```

Comment:
I think counting the ellipsis should be in the limit like you did? so in case of custom ellipsis it preserves the total length 🤔

Reply:
I followed the initial pattern that was in the function, but can do.

Reply:
done with this